packages/web/src/pages/dashboard-page/components/ArtistContentSection.tsx

```diff
@@ -0,0 +1,461 @@
+      (data) =>
+        data.is_unlisted === false &&
+        !isContentUSDCPurchaseGated(data.stream_conditions)
+    )
+    const specialAccessTracks = tracks.filter(
+      (data) =>
+        'tip_user_id' in (data.stream_conditions || {}) ||
+        'follow_user_id' in (data.stream_conditions || {})
+    )
+    const hiddenTracks = tracks.filter((data) => data.is_unlisted === true)
+    const premiumTracks = tracks.filter(
+      (data) => 'usdc_purchase' in (data.stream_conditions || {})
+    )
+    const collectibleGatedTracks = tracks.filter(
+      (data) => 'nft_collection' in (data.stream_conditions || {})
+    )
+
+    const arrays = [
+      publicTracks,
+      specialAccessTracks,
+      hiddenTracks,
+      premiumTracks,
+      collectibleGatedTracks
+    ]
+    const nonEmptyArrays = arrays.filter((arr) => arr.length > 0)
+    const hasOnlyOneTrackSection = nonEmptyArrays.length === 1
+
+    return {
+      hasOnlyOneTrackSection,
+      publicTracks,
+      specialAccessTracks,
+      hiddenTracks,
+      premiumTracks,
+      collectibleGatedTracks
+    }
+  }, [tracks])
+
+  const { hasOnlyOneAlbumSection, publicAlbums, hiddenAlbums, premiumAlbums } =
+    useMemo(() => {
+      const publicAlbums = albums.filter(
+        (data) =>
+          data.is_private === false &&
+          !isContentUSDCPurchaseGated(data.stream_conditions)
+      )
+      const hiddenAlbums = albums.filter((data) => data.is_private === true)
+      const premiumAlbums = albums.filter(
+        (data) => 'usdc_purchase' in (data.stream_conditions || {})
+      )
+
+      const arrays = [publicAlbums, hiddenAlbums, premiumAlbums]
+      const nonEmptyArrays = arrays.filter((arr) => arr.length > 0)
+      const hasOnlyOneAlbumSection = nonEmptyArrays.length === 1
+
+      return {
+        hasOnlyOneAlbumSection,
+        publicAlbums,
+        hiddenAlbums,
+        premiumAlbums
+      }
+    }, [albums])
+
+  let filteredTracks: DataSourceTrack[] = tracks
+  let filteredAlbums: DataSourceAlbum[] = albums
+
+  switch (selectedTrackFilter) {
+    case TrackFilters.PUBLIC:
+      filteredTracks = publicTracks
+      break
+    case TrackFilters.PREMIUM:
+      filteredTracks = premiumTracks
+      break
+    case TrackFilters.SPECIAL_ACCESS:
+      filteredTracks = specialAccessTracks
+      break
+    case TrackFilters.COLLECTIBLE_GATED:
+      filteredTracks = collectibleGatedTracks
+      break
+    case TrackFilters.HIDDEN:
+      filteredTracks = hiddenTracks
+      break
+    default:
+      filteredTracks = tracks
+      break
+  }
+  switch (selectedAlbumFilter) {
+    case AlbumFilters.PUBLIC:
+      filteredAlbums = publicAlbums
+      break
+    case AlbumFilters.PREMIUM:
+      filteredAlbums = premiumAlbums
+      break
+    case AlbumFilters.HIDDEN:
+      filteredAlbums = hiddenAlbums
+      break
+    default:
+      filteredAlbums = albums
+      break
+  }
+
+  const filterButtonOptions = useMemo(() => {
```

Comment:
Some of the logic in this component could probably be pulled out into helper functions to make this easier to read.